lib/get-format.js

```diff
@@ -18,7 +18,10 @@ const extensionFormatMap = {
   '.cjs': 'commonjs',
   '.js': 'module',
   '.json': 'json',
-  '.mjs': 'module'
+  '.mjs': 'module',
+  '.ts': 'module',
+  '.cts': 'commonjs',
+  '.mts': 'module'
```

Comment:
I'd be _very surprised_ if node hard-codes TS-specific extensions. Doesn't it have a generic fallback for any unknown file type?

Reply:
It looks like the fix should go in the `getFileProtocolModuleFormat`, where Node.js implements the fallback:
https://github.com/nodejs/node/blob/40fa2e9c110c616b1301c20c2a3cdb2cf6cc165e/lib/internal/modules/esm/get_format.js#L83

Reply:
It looks like the fix should go in the getFileProtocolModuleFormat, where Node.js has an additional ignoreError parameter to suppress that error:
[nodejs/node@40fa2e9/lib/internal/modules/esm/get_format.js#L66-L67](https://github.com/nodejs/node/blob/40fa2e9c110c616b1301c20c2a3cdb2cf6cc165e/lib/internal/modules/esm/get_format.js#L66-L67)

Reply:
And that parameter is explicitly set to `true`: https://github.com/nodejs/node/blob/40fa2e9c110c616b1301c20c2a3cdb2cf6cc165e/lib/internal/modules/esm/get_format.js#L107-L111

Reply:
Yeah makes sense. This is the patch I applied to get things working in my local branch. I'll take a closer look.

Reply:
Thanks for working on a fix! :)
The extension map is indeed *not supposed to be changed*.
Nicolò’s comment points to `getFileProtocolModuleFormat`, which in Node receives an `ignoreErrors` and `returns undefined` if that is passed.
That should be accepted.
Furthermore, it should be passed as `true` from `defaultGetFormatWithoutErrors`.

Reply:
@nicolo-ribaudo I've made modifications to better reflect the way node is doing things (based on the links provided).